src/boss/rbo3/rbo3.c

```diff
@@ -1,13 +1,500 @@
+                }
+                break;
+
+            case 7:
+                if (g_PlayerX < 128) {
+                    g_Tilemap.x = 128;
+                    self->step++;
+                }
+                break;
+
+            case 8:
+                if (g_PlayerX > 640) {
+                    g_Tilemap.width = 640;
+                    self->step++;
+                }
+                break;
+
+            case 9:
+                if (g_PlayerX < 128) {
+                    g_Tilemap.x = 128;
+                    self->step++;
+                }
+                break;
+
+            case 10:
+                if (g_PlayerX > 640) {
+                    g_Tilemap.width = 640;
+                    g_Tilemap.right--;
+                    self->step++;
+                }
+                break;
+
+            case 11:
+                if (g_PlayerX < 384) {
+                    g_Tilemap.x = 384;
+                    g_Tilemap.left++;
+                    self->step++;
+                }
+                break;
+
+            case 12:
+                if (g_PlayerX > 640) {
+                    g_Tilemap.width = 640;
+                    g_Tilemap.right--;
+                    self->step++;
+                }
+                break;
+
+            case 13:
+            case 14:
+                if (g_PlayerX < 256) {
+                    g_Tilemap.x = 256;
+                    g_Tilemap.left++;
+                    self->step++;
+                }
+                break;
+            }
+        }
+        break;
+    }
+}
+
+extern Entity D_8007A958;
+extern Entity D_8007C0D8;
+extern u16 D_us_80180480;
+extern u8 D_us_801805F0;
+extern u8 D_us_801805FC;
+extern u8 D_us_8018060C;
+extern u8 D_us_80180618;
+extern u8 D_us_80180624;
+extern u8 D_us_80180630;
+extern u8 D_us_8018063C;
+extern s8 D_us_80180648[];
+extern u8 D_us_80180670[];
+extern s32 D_us_80180728;
+
+void EntityMedusa(Entity* self) {
+    Entity* entity;
+    s16 rotZ;
+    s32 x;
+    s32 i;
+    s32 velocityX;
+    s8* rect;
+    u8* indexes;
+
+    if (self->flags & 0x100) {
+        if (self->step != 7) {
+            SetStep(7);
+        }
+    }
+
+    switch (self->step) {
+    case 0:
+        InitializeEntity(&D_us_80180480);
+        self->animCurFrame = 1;
+        self->hitboxState = 0;
+        CreateEntityFromEntity(UNK_ENTITY_25, self, self + 1);
+        SetStep(1);
+        // fallthrough
+
+    case 1:
+        if (D_us_80180728 & 1) {
+            SetStep(2);
+        }
+        break;
+
+    case 2:
+        // n.b.! AnimateEntity is not declared
+        if (!AnimateEntity(&D_us_801805F0, self)) {
+            self->hitboxState = 3;
+            SetStep(3);
+        }
+        break;
+
+    case 3:
+        AnimateEntity(&D_us_801805FC, self);
+        if (self->step_s == 0) {
+            self->ext.GS_Props.timer = 64;
+            self->step_s++;
+        }
+        if (GetDistanceToPlayerX() < 104) {
+            self->ext.GS_Props.attackMode = 1;
```

Comment:
My theory is that bosses will use similar extension layouts., but I think it's too early to commit and rename `GB_Props`. This is intended to look out of place to be resolved when more is known about these layouts.